scan-command/src/main/java/io/ballerina/scan/internal/CoreRules.java

```diff
@@ -19,23 +19,23 @@
 package io.ballerina.scan.internal;
 
 import io.ballerina.scan.Rule;
-
-import java.util.ArrayList;
-import java.util.List;
+import io.ballerina.scan.RuleKind;
 
 /**
  * {@code CoreRules} contains the core static code analysis rules.
  *
  * @since 0.1.0
  * */
-public class CoreRules {
-    private final List<Rule> coreRules;
+enum CoreRules {
+    RULE_CHECKPANIC(RuleFactory.createRule(1, "Avoid checkpanic", RuleKind.CODE_SMELL));
+
+    private final Rule rule;
 
-    CoreRules() {
-        coreRules = new ArrayList<>();
+    CoreRules(Rule rule) {
+        this.rule = rule;
     }
 
-    List<Rule> getCoreRules() {
-        return coreRules;
+    public Rule rule() {
+        return rule;
     }
```

Comment:
The enum should probably be name CoreRule.

Also, can't we add a method here to get List<CoreRule>?